tests/iostability/test_iostability_degraded_path.py

```diff
@@ -116,3 +122,60 @@ def test_object_crud_single_pod_failure(self):
         self.test_completed = True
         self.log.info("ENDED: Test for Object CRUD operations in degraded mode in loop using "
                       "S3bench for 7 days")
+
+    @pytest.mark.lc
+    @pytest.mark.io_stability
+    @pytest.mark.tags("TEST-40173")
+    def test_disk_near_full_read_in_degraded_s3bench(self):
+        """
+        Perform disk storage near full once in healthy cluster and read in degraded cluster in loop
+        for 7 days.
+        """
+        self.log.info("STARTED: Perform disk storage near full once in healthy cluster and "
+                      "read in degraded cluster in loop for 7 days.")
+        self.log.info("Step 1: calculating byte count for required percentage")
+        resp = self.dfr.get_user_data_space_in_bytes(master_obj=self.master_node_list[0],
+                                                     memory_percent=95)
+        if resp[1] != 0:
+            self.log.info("Need to add %s for required percentage", resp[1])
+            self.log.info("Step 2: performing writes till we reach required percentage")
+            s3userinfo = dict()
+            s3userinfo['accesskey'] = ACCESS_KEY
+            s3userinfo['secretkey'] = SECRET_KEY
+            bucket_prefix = "testbkt"
+            ret = DiskFailureRecoveryLib.perform_near_full_sys_writes(s3userinfo=s3userinfo,
+                                                                      user_data_writes=int(resp[1]),
+                                                                      bucket_prefix=bucket_prefix,
+                                                                      client=20)
+            if ret[0]:
+                assert False, "Errors in write operations."
+            else:
+                self.log.debug("write operation data: %s", ret)
+                self.log.info("Step 2: Shutdown the data pod safely by making replicas=0, "
+                              "check degraded status.")
+                resp = self.ha_obj.delete_kpod_with_shutdown_methods(self.master_node_list[0],
+                                                                     self.hlth_master_list[0])
+                assert_utils.assert_true(resp[0], resp[1])
+                self.log.info("Deleted pod : %s", resp[1])
```

Comment:
resp[1] of this method is a dict, maybe u need to get the pod name
pod_name = list(resp[1].keys())[0]

Reply:
fixed